### ISSUE TITLE
Coregister fails for non-overlapping domains

### ISSUE THREAD

**Issue by thomas-eldridge** [CCI-Tools/cate#495]
### Expected behavior

Downloaded/ opened two datasets covering identical temporal range but different geospatial regions (i.e. reduced download size while exploring teleconnection). Expected to be able to coregister the datasets (to same lat/lon granularity) despite lack of geospatial overlap.

### Actual behavior

Error raised by Cate, presumably because latitude/ longitude domains do not overlap

### Steps to reproduce the problem

1. Download/open SST ds for 5S - 5N, 170W - 120W (NINO3.4 region)
2. Download/open soil moisture ds for Northern AUS/ Malaysian peninsula (because known teleconnection w. ENSO)
3. Attempt coregister

### Specifications 

Cate v1.0 dev 3

**Comment by forman** [comment]
Does the same apply to Cate **1.0**? 
Note that version **1.0 dev 3** is outdated.

**Comment by JanisGailis** [comment]
Yes, the ``coregistration`` operation has been defined [Coregistration op specification](https://cate.readthedocs.io/en/latest/op_specs/geometric-adjustments/op_spec_coregistration.html) in the context of Use Case 9 to enable pixel-by-pixel analysis of two datasets observing the same spatial scene. Hence the grids include spatial extents as well, and even do cropping if necessary.

@thomas-eldridge Is pixel by pixel analysis important to explore teleconnection?
@forman It should be possible to change how the operation works by relying on pixel size alone to determine width and height of the slave dataset, such that regions that don't overlap can be coregistered. It will mean that how coregistration operation works will change, so on top of that we might need to implement ``subset_as`` to preserve the current functionality. What do you think?

**Comment by forman** [comment]
I'd prefer, if `coregistration` worked like @thomas-eldridge is expecting it to work. Btw, this is also how the SNAP `collocate` operator is working. 

What about a new keyword argument `allow_disjoint_inputs: bool = False`.

**Comment by JanisGailis** [comment]
@forman In your opinion, should coregistration crop the slave to master as it does now?
